### DISCUSSION TITLE
[Feature Request] support <Icon> Component

### DISCUSSION THREAD

**Discussion by trydofor** [vuepress-theme-hope/vuepress-theme-hope#1845] [Q&A] [answered]
### Describe the feature

like `<Badge>` component, support `<Icon>` to show iconfont/fontawesome

**Comment by Mister-Hope** [comment]
Unlike `<Badge>` is standalone with theme, `<Icon>` is coupled with theme, and needs global configuration.

Also I personally not found this useful, anyway thanks for your idea, and I extract the icon logic in to a component in https://github.com/vuepress-theme-hope/vuepress-theme-hope/commit/7b14c2d6d5fd53a742570955ab5e69f58c5335ea and you should be able to import `@theme-hope/components/Icon` and register it globally by your own, see https://vuepress-theme-hope.github.io/v2/cookbook/advanced/component.html for more details.

**Reply by trydofor** [reply]
> Also I personally not found this useful,

inline `Icon` has the same purpose as `Badge` and Emoji.

> `<Icon>` is coupled with theme

and just expose it will be happy

**Comment by trydofor** [comment]
temporarily use `@vuepress/plugin-register-components` register a Component like following
```vue
<template>
  <i :class="['icon',  iconPrefix + icon]"></i>
</template>

<script setup lang="ts">
import { defineProps } from 'vue';
import { themeOption } from '../configs/theme';

defineProps<{
  icon: string;
}>();

const iconPrefix = themeOption.iconPrefix || '';
</script>

<style scoped>
</style>
```

**Reply by Mister-Hope** [reply]
DO NOT USE THIS.

You can not import vuepress config file as it's working on node side, by doing this, you will include all node side code in your app, and it may break your application.

**Reply by trydofor** [reply]
Thanks !

i only use the `type` to define a plain object options, is this ok?

```ts
import { HopeThemeOptions } from 'vuepress-theme-hope';

export const themeOption = {
  hostname: 'https://java-code-review.moilioncircle.com',
  author: {
    name: 'trydofor',
    url: 'https://www.trydofor.com',
  },
  // others ....
} as HopeThemeOptions;
```

**Reply by Mister-Hope** [reply]
That's ok., but you should use `useThemeData` to get themedata.

**Reply by trydofor** [reply]
thx, and waiting for https://github.com/vuepress-theme-hope/vuepress-theme-hope/commit/7b14c2d6d5fd53a742570955ab5e69f58c5335ea release

**Comment by Mister-Hope** [comment, marked as answer]
`<FontIcon>` is supported since beta.67

**Reply by trydofor** [reply]
thanks very much, bump to beta71.
and no need global register, just use `<FontIcon icon="speed"/>`
